spvc/test/run_spirv_cross_tests.py

```diff
@@ -45,127 +43,112 @@
 test_count = 0
 pass_count = 0
 
-not_used, tmpfile = tempfile.mkstemp()
-devnull = open(os.devnull, 'w')
-
-def mkdir_p(filepath):
-    ''' make sure directory exists to contain given file '''
-    dirpath = os.path.dirname(filepath)
-    try:
-        os.makedirs(dirpath)
-    except OSError as exc:
-        if not (exc.errno == errno.EEXIST and os.path.isdir(dirpath)):
-            raise
+def check_call(cmd):
+    global devnull
+    if subprocess.call(cmd, stdout=devnull):
+        print('failed command: ', ' '.join(cmd))
+        sys.exit(1)
 
-def spirv_as(inp, out):
-    subprocess.check_call([spirv_as_path, '-o', out, inp], stdout=devnull)
+def spirv_as(inp, out, flags):
+    check_call([spirv_as_path] + flags + ['-o', out, inp])
 
-def spirv_opt(inp, out):
-    subprocess.check_call([spirv_opt_path, '-o', out, inp], stdout=devnull)
+def spirv_opt(inp, out, flags):
+    check_call([spirv_opt_path] + flags + ['--skip-validation', '-O', '-o', out, inp])
 
-def glslangValidator(inp, out):
-    subprocess.check_call([glslangValidator_path , '--target-env', 'vulkan1.1', '-V', '-o',
-                          out, inp], stdout=devnull)
+def glslang_compile(inp, out, flags):
+    check_call([glslangValidator_path] + flags + ['-o', out, inp])
 
 def spvc(inp, out, flags):
-    return subprocess.call([spvc_path, '-o', out, '--validate=vulkan1.1'] + flags + [inp],
-                           stdout=devnull)
-
-def compile_shader(source_path, binary_path, filename, optimize):
-    # Note: filename is the last component of source_path.
-    mkdir_p(binary_path)
-    if '.asm.' in filename:
-        spirv_as(source_path, binary_path)
-    else:
-        glslangValidator(source_path, binary_path)
-    if optimize:
-        spirv_opt(binary_path, binary_path)
-
-def do_test(binary_path, reference_path, flags):
-    global test_count, pass_count
+    global devnull
+    global test_count
     test_count += 1
-    if spvc(binary_path, tmpfile, flags) == 0 and filecmp.cmp(tmpfile, reference_path):
+    cmd = [spvc_path] + flags + ['-o', out, '--validate=vulkan1.1', inp]
+    return subprocess.call(cmd, stdout=devnull)
+
+def check_reference(result, reference):
+    global spirv_cross_dir
+    global pass_count
+    if result and filecmp.cmp(result, os.path.join(spirv_cross_dir, reference), False):
         pass_count += 1
 
-def skip_test():
-    global test_count
-    test_count += 1
+def remove_files(*filenames):
+    for i in filenames:
+        try:
+            os.remove(i)
+        except:
+            pass
+
+def test_glsl(shader, filename, optimize):
```

Comment:
Very much needs a method comment.

This method name has glsl in it, but it's definitely handling SPIR-V assembly cases, etc.
Badly named method?

Reply:
This script or this method should describe that it's tracking the behaviours of SPIRV-Cross' own testing tactics.  Give the next person a place to look in SPIRV-Cross itself when there's new or changed behaviour to be replicated.

Reply:
I expanded the comments.  The function could be split up, but there's also something to be said for not having to page up and down looking for functions.  Instead I've tried to clearly label the three main steps with comments.